climate-tagger.php

```diff
@@ -12,117 +12,149 @@
 	define( 'CLIMATE_TAGGER_INCLUDES', WP_PLUGIN_URL . '/climate-tagger' );
 }
 
+// Documentation: http://api.climatetagger.net/documentation/
+define( 'CLIMATE_TAGGER_API_URL', 'http://api.climatetagger.net' );
+
 class ClimateTagger {
 
-	function admin_menu() {
+	static function admin_menu() {
 		add_options_page(
 			'Climate Tagger',
 			'Climate Tagger',
 			'manage_options',
 			'climate-tagger',
-			array( 'ClimateTagger', 'add_options_page_callback' ));
+			array( 'ClimateTagger', 'add_options_page_callback' )
+		);
 
 	}
 
-	function admin_init()
-	{
+	static function admin_init() {
 		self::set_defaults();
 
 		register_setting(
 			'climate_tagger_general_settings',
 			'climate_tagger_general_settings'
 		);
-	}
 
-	function set_defaults() {
+		// Add an action link pointing to the settings page.
+		add_filter( 'plugin_action_links_' . plugin_basename( __FILE__ ), array(
+			'ClimateTagger',
+			'add_action_links',
+		) );
+	}
 
+	static function set_defaults() {
 		$options = get_option( 'climate_tagger_general_settings' );
 
 		$options = wp_parse_args(
 			$options,
 			array(
-				'token' => '',
-				'limit' => '20',
+				'token'      => '',
+				'limit'      => '20',
 				'post_types' => 'post',
-			) );
+				'project'    => 'default',
```

Comment:
You can simply remove `'project' => 'default'` now because if the plugin is upgraded and the project setting is not changed by the admin, you will end up sending `'default'` as the tagger to the API, which will result in an internal server error being returned.